File: export_data.py
#!/usr/bin/env python3
import os
import json
from supabase import create_client, Client

url = os.getenv("VITE_SUPABASE_URL")
key = os.getenv("VITE_SUPABASE_ANON_KEY")
supabase: Client = create_client(url, key)

output_dir = "/tmp/cc-agent/56455764/project/migration-export-data"
os.makedirs(output_dir, exist_ok=True)

tables = {
    "users": "id, name, email, role, created_at",
    "projects": "*",
    "stages": "*",
    "stage_items": "*",
    "item_checks": "*",
    "stage_statuses": "*",
    "notifications": "*",
    "attachments": "*",
    "activity_log": "*",
    "project_costs": "*",
    "project_variations": "*"
}

manifest = []

print("Starting data export...")

for table_name, columns in tables.items():
    print(f"\nExporting {table_name}...")

    try:
        response = supabase.table(table_name).select(columns).execute()
        data = response.data

        filename = f"{table_name}.json"
        filepath = os.path.join(output_dir, filename)

        with open(filepath, 'w') as f:
            json.dump(data, f, indent=2, default=str)

        row_count = len(data)
        file_size = os.path.getsize(filepath)

        manifest.append({
            "table": table_name,
            "filename": filename,
            "row_count": row_count,
            "file_size_bytes": file_size,
            "format": "JSON",
            "status": "✅ Complete"
        })

        print(f"  ✅ {table_name}: {row_count} rows exported ({file_size:,} bytes)")

    except Exception as e:
        print(f"  ❌ Error exporting {table_name}: {str(e)}")
        manifest.append({
            "table": table_name,
            "filename": filename,
            "row_count": 0,
            "file_size_bytes": 0,
            "format": "JSON",
            "status": f"❌ Error: {str(e)}"
        })

with open(os.path.join(output_dir, "EXPORT_MANIFEST.json"), 'w') as f:
    json.dump({
        "export_date": "2026-04-08",
        "source_database": "izufnkvcdbshjuwuxhhr",
        "total_tables": len(manifest),
        "tables": manifest
    }, f, indent=2)

print("\n" + "="*60)
print("EXPORT COMPLETE")
print("="*60)
print(f"\nExport location: {output_dir}")
print(f"Total tables exported: {len(manifest)}")
print(f"Total rows exported: {sum(m['row_count'] for m in manifest)}")
print("\nManifest created: EXPORT_MANIFEST.json")

File: export-data.mjs
import { createClient } from '@supabase/supabase-js';
import fs from 'fs';
import path from 'path';

const url = process.env.VITE_SUPABASE_URL;
const key = process.env.VITE_SUPABASE_ANON_KEY;
const supabase = createClient(url, key);

const outputDir = '/tmp/cc-agent/56455764/project/migration-export-data';
if (!fs.existsSync(outputDir)) {
  fs.mkdirSync(outputDir, { recursive: true });
}

const tables = [
  'users',
  'projects',
  'stages',
  'stage_items',
  'item_checks',
  'stage_statuses',
  'notifications',
  'attachments',
  'activity_log',
  'project_costs',
  'project_variations'
];

const manifest = [];

console.log('Starting data export...\n');

for (const tableName of tables) {
  console.log(`Exporting ${tableName}...`);

  try {
    const { data, error } = await supabase
      .from(tableName)
      .select('*');

    if (error) throw error;

    const filename = `${tableName}.json`;
    const filepath = path.join(outputDir, filename);

    fs.writeFileSync(filepath, JSON.stringify(data, null, 2));

    const stats = fs.statSync(filepath);
    const rowCount = data.length;

    manifest.push({
      table: tableName,
      filename: filename,
      row_count: rowCount,
      file_size_bytes: stats.size,
      format: 'JSON',
      status: '✅ Complete'
    });

    console.log(`  ✅ ${tableName}: ${rowCount} rows exported (${stats.size.toLocaleString()} bytes)`);

  } catch (error) {
    console.log(`  ❌ Error exporting ${tableName}: ${error.message}`);
    manifest.push({
      table: tableName,
      filename: `${tableName}.json`,
      row_count: 0,
      file_size_bytes: 0,
      format: 'JSON',
      status: `❌ Error: ${error.message}`
    });
  }
}

const manifestData = {
  export_date: new Date().toISOString().split('T')[0],
  source_database: 'izufnkvcdbshjuwuxhhr',
  total_tables: manifest.length,
  total_rows: manifest.reduce((sum, m) => sum + m.row_count, 0),
  total_size_bytes: manifest.reduce((sum, m) => sum + m.file_size_bytes, 0),
  tables: manifest
};

fs.writeFileSync(
  path.join(outputDir, 'EXPORT_MANIFEST.json'),
  JSON.stringify(manifestData, null, 2)
);

console.log('\n' + '='.repeat(60));
console.log('EXPORT COMPLETE');
console.log('='.repeat(60));
console.log(`\nExport location: ${outputDir}`);
console.log(`Total tables exported: ${manifest.length}`);
console.log(`Total rows exported: ${manifestData.total_rows.toLocaleString()}`);
console.log(`Total size: ${(manifestData.total_size_bytes / 1024 / 1024).toFixed(2)} MB`);
console.log('\nManifest created: EXPORT_MANIFEST.json');
